src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -279,13 +295,22 @@ private static NuGetFramework GetToolFramework(IEnumerable targetFrameworks)
             IEnumerable values,
             string propertyName,
             Func<string, TValue> valueFactory)
+        {
+            return GetNonEvaluatedPropertyOrNull(values, propertyName, valueFactory, default(TValue));
+        }
+
+        private static IEnumerable<TValue> GetNonEvaluatedPropertyOrNull<TValue>(
+            IEnumerable values,
+            string propertyName,
+            Func<string, TValue> valueFactory,
+            TValue defaultValue)
```

Comment:
You don't need a default value, the value factory can handle that.

Reply:
Updated and removed the default approach.